tests/post/steps/test_solutions.py

```diff
@@ -0,0 +1,350 @@
+    run_solutions_command(
+        request,
+        host,
+        ["delete-solution", "--name", environment, "--solution", name]
+    )
+
+
+@when(parsers.parse("we delete Solution environment '{environment}'"))
+def delete_solution_env(host, request, environment):
+    run_solutions_command(request, host, ["delete-env", "--name", environment])
+
+
+@when(parsers.parse("we unimport Solution archive '{path}'"))
+def unimport_solution_archive(host, request, path):
+    run_solutions_command(request, host, ["unimport", "--archive", path])
+
+
+@when("we delete the Solution Configuration file")
+def remove_solution_config(host, request):
+    with host.sudo():
+        res = host.run(('rm -rf {}').format(SOLUTION_CONFIGURATION_FILE))
+        assert res.rc == 0, res.stdout
+
+# }}}
+
+
+# Then {{{
+
+@then(parsers.parse("Solution archive '{name}' is imported correctly"))
+def is_imported_solution_archive(host, k8s_client, name):
+    # Solution is imported correctly if
+    # 1. Solution Configuration is created at /etc/metalk8s/solution.yaml
+    # 2. Solution Configuration file has new archive path
+    # 3. Solutions ConfigMap has the new Solution imported
+    with host.sudo():
+        assert host.file(SOLUTION_CONFIGURATION_FILE).exists, \
+            'No Solution Configuration file exists at {}'.format(
+                SOLUTION_CONFIGURATION_FILE
+            )
+    # Todo: SOLUTION_CONFIGURATION_FILE is yaml, we could parse it
+    # and ensure the archive path is at the right location
+    with host.sudo():
+        assert name in host.file(SOLUTION_CONFIGURATION_FILE).content_string, \
+            'No {} Solution ISO path found in Solution Config {}'.format(
+                name, SOLUTION_CONFIGURATION_FILE
+            )
+
+    def _wait_for_imported_archive():
+
+        configmap = get_configmap(
+            k8s_client, SOLUTION_CONFIGMAP, SOLUTION_NAMESPACE
+        )
+        assert configmap is not None, \
+            'Failed to read Solution ConfigMap {}'.format(SOLUTION_CONFIGMAP)
+
+        data = configmap.get('data', {})
+        assert any(k == name for k, v in data.items()), \
+            '{} Configmap does not contain solution {}'.format(
+                SOLUTION_CONFIGMAP, name
+            )
+    utils.retry(
+        _wait_for_imported_archive,
+        times=10,
+        wait=5,
+        name="wait for imported archive"
+    )
+
+
+@then(parsers.parse(
+    "Solution '{name}' version '{solution_version}' is available"))
+def is_available_solution(host, request, name, solution_version):
+    available_solution = get_available_solution(host)
+    assert name in available_solution
+    for k, v in available_solution.items():
+        if k == name:
+            for item in v:
+                assert item.get('version', None) == solution_version, \
+                    'Expected solution version {}: got {}'.format(
+                        solution_version, item.get('version', None)
+                    )
```

Comment:
Not the good way of doing this, you should check each item to see if there is a matching version.